spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDStreamParser.java

```diff
@@ -354,20 +385,27 @@ public DocumentParseResult parse(String[] document) {
 			DocumentParseResult result = new DocumentParseResult(document.length);
 			CrudRepository<BaseDefinition, String> transientRepository = new TransientDefinitionRepository();
 			StreamConfigParser parser = new StreamConfigParser(transientRepository);
+			int line = 1;
 			for (String nameAndDefinitionPair : document) {
 				try {
-
 					StreamNode stream = parser.parse(nameAndDefinitionPair);
 					String streamName = stream.getStreamName();
+					if (streamName == null) {
+						// Give the stream a placeholder name as none was supplied. Someone
+						// processing the parse result should know this can happen.
+						streamName = "UNKNOWN_" + line;
+					}
+					List<Exception> errorAccumulator = new ArrayList<Exception>();
 					List<ModuleDescriptor> moduleDescriptors = delegate.buildModuleDescriptors(streamName,
-							nameAndDefinitionPair, ParsingContext.stream, stream);
+							nameAndDefinitionPair, ParsingContext.partial_stream, stream, errorAccumulator);
 					BaseDefinition streamDefinition = new StreamDefinition(streamName, nameAndDefinitionPair);
 					transientRepository.save(streamDefinition);
-					result.success(moduleDescriptors);
+					result.success(moduleDescriptors, errorAccumulator);
```

Comment:
There are two kinds of problems that can occur when parsing a definition. A syntax error or a validation error. On a syntax error I'm not really expecting a structure back.  On a validation error there is really nothing wrong with the parsed structure and it can be used by the client to graphically show the stream and then we can highlight the pieces of the stream with any validation errors. If all the get is the validation errors we can't show the user any kind of up to date graph. This becomes a shame when a user is typing because we want to reduce the times we can't show a graph.  As a user starts typing his stream 'mail', when he presses 'm' we can immediately show a node in the graph marked with 'not a known module', as they continue typing to spell 'mail' the graph constantly updates and eventually is shown as a valid mail node with no validation errors. Suppose they then start extending it 'mail | tran' at this point, there is a validation error on 'tran' (they haven't finished typing transform') but unless you give me the parsed structure, should I blank out the graph? the 'mail' node would then disappear when we know that is OK.  Hope you get what I'm describing.